### PR TITLE
[prim_present/otp_ctrl] Add round index state IOs to primitive

### PR DESCRIPTION
**Pull request by msfschaffner**
This addresses #3784.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Review by udinator — APPROVED** [review]
the changes to prim_present design and testbench LGTM - once this is merged I'll enhance the testbench to verify that the new outputs are correct. 

@msfschaffner I think explicitly testing the iterative decomposition is unnecessary since it's implied that if an N-round encryption is correct then a decomposed algorithm will be correct assuming that correct keys are provided at each stage, wdyt?

**Comment by msfschaffner** [comment]
Hey @udinator, this probably depends a bit how coverage will be affected by this. But from my side I think it is ok to just test the static version.

File: hw/ip/prim/rtl/prim_present.sv
Comment (cindychip) [inline hw/ip/prim/rtl/prim_present.sv:85]: Hey michael, thanks for the quick fix. Now I can see the encryption key is correct, but the decryption key does not match the original value. I think it might related to the `NumRounds` here? In OTP this is set to 1.

Comment (msfschaffner) [reply hw/ip/prim/rtl/prim_present.sv]: Right, sorry, my solution was only half baked. How does it look now?

Also, I realized that I had already implemented a `prim_cipher_pkg::present_get_dec_key128` function that we can leverage to simplify the LUT creation...

Comment (cindychip) [reply hw/ip/prim/rtl/prim_present.sv]: lgtm! the sanity test passed

**Review by cindychip — APPROVED** [review]

**Pull request merged**
